examples/fuse_examples/imaging/classification/knight/baseline/dataset.py

```diff
@@ -1,198 +1,264 @@
+        if bmi!=None and bmi > 10 and bmi < 100:
+            bmi = np.array(bmi / 50.0).reshape(-1)
+        else:
+            bmi = np.array(-1.0).reshape(-1)
+
+        radiographic_size = sample_dict['data.input.clinical.radiographic_size']
+        if radiographic_size!=None and radiographic_size > 0 and radiographic_size < 50:
+            radiographic_size = np.array(radiographic_size / 15.0).reshape(-1)
+        else:
+            radiographic_size = np.array(-1.0).reshape(-1)
+        
+        preop_egfr = sample_dict['data.input.clinical.last_preop_egfr']
+        if preop_egfr!=None and preop_egfr > 0 and preop_egfr < 200:
+            preop_egfr = np.array(preop_egfr / 90.0).reshape(-1)
+        else:
+            preop_egfr = np.array(-1.0).reshape(-1)
+        # turn categorical features into one hot vectors
+        gender = sample_dict['data.input.clinical.gender']
+        gender_one_hot = np.zeros(len(GENDER_INDEX))
+        if gender in GENDER_INDEX.values():
+            gender_one_hot[gender] = 1
+
+        comorbidities = sample_dict['data.input.clinical.comorbidities']
+        comorbidities_one_hot = np.zeros(len(COMORBIDITIES_INDEX))
+        if comorbidities in COMORBIDITIES_INDEX.values():
+            comorbidities_one_hot[comorbidities] = 1
+        
+        smoking_history = sample_dict['data.input.clinical.smoking_history']
+        smoking_history_one_hot = np.zeros(len(SMOKE_HISTORY_INDEX))
+        if smoking_history in SMOKE_HISTORY_INDEX.values():
+            smoking_history_one_hot[smoking_history] = 1
+        
+
+        clinical_encoding = np.concatenate((age, bmi, radiographic_size, preop_egfr, gender_one_hot, comorbidities_one_hot, smoking_history_one_hot), axis=0, dtype=np.float32)
+        sample_dict["data.input.clinical.all"] = clinical_encoding
+        return sample_dict
 
 def knight_dataset(data_dir: str = 'data', cache_dir: str = 'cache', split: dict = None, \
         reset_cache: bool = False, \
-        rand_gen = None, batch_size=8, resize_to=(256,256,110), task_num=1, \
+        rand_gen = None, batch_size=8, resize_to=(110,256,256), task_num=1, \
         target_name='data.gt.gt_global.task_1_label', num_classes=2, only_labels=False):
+    
+
+    static_pipeline = PipelineDefault("static", [
+        # decoding sample ID
+        (OpKnightSampleIDDecode(), dict()), # will save image and seg path to "data.input.img_path", "data.gt.seg_path" and load json data
+        (OpClinicalLoad(os.path.join(data_dir, 'knight.json')), dict()),
+        # loading data
+        (OpLoadImage(data_dir), dict(key_in="data.input.img_path", key_out="data.input.img", format="nib")),
+        # (OpLoadImage(data_dir), dict(key_in="data.gt.seg_path", key_out="data.gt.seg", format="nib")),
+        
+        
+        # fixed image normalization
+        (OpClip(), dict(key="data.input.img", clip=(-62, 301))),
+        (OpLambda(lambda x: (x - 104.0)/75.3 ), dict(key="data.input.img")), #kits normalization
```

Comment:
Can you implement OpZScoreNorm for this one (seems like a useful op)

Reply:
done